src/main/java/com/microsoft/alm/authentication/KeychainSecurityCliStore.java

```diff
@@ -304,6 +307,22 @@ else if ("blob".equals(type.toString()))
         // TODO: else if ("sint32".equals(type))
     }
 
+    public boolean isKeychainAvailable()
+    {
+        final String targetName = "test:isKeychainAvailable";
+        final Token token = new Token("this is a test token", TokenType.Test);
+        try
+        {
+            writeToken(targetName, token);
+        }
+        catch (final SecurityException e)
+        {
+            return false;
+        }
+        delete(targetName);
+        return true;
+    }
```

Comment:
In isKeychainAvailable(), is there a better way to check availability of the keychain than an attempted write and delete?  Just wondering if a different 'security' command line argument might quickly reveal the same thing, or a system call via JNA.  Perhaps the write/delete is fine.